chocolatey/choco
# (#158) Split out list command to its own command class

## Description Of Changes

This pull requests splits out the list command to be its own command in the source codes, which can only handle local packages and not remote sources.
In this move, all arguments that had been deprecated for the list command have been completely removed from the list command, and a few remote only arguments will be warning about that they are unsupported (this won't do anything at all).

The tab completion file has also been updated with this new logic, with an adition that remote searches have had their local arguments removed from the tab completion.

## Motivation and Context

To make list searching only local packages, while using search and find will be for remote sources.

## Testing

1. Run `choco list`
2. Verify only locally installed packages is shown
3. Run `choco search` against a feed not containing 1 or more of the installed pakcages
3. Verify the packages only available locally is not shown
4. Run `choco list --local-only`
5. Verify a warning is outputted (purple) about the argument being unsupported
6. Run `choco search --local-only --verbose`
7. Verify the console mentions it used a query with the search term `--local-only`
9. Run `choco info chocolatey --local-only`
10. Verify information is shown about the local package, and no warning about unsupported argument.
11. Run `choco info upgradepackage` (against internal repository source)
12. Verify the remote package information is shown
12. (Optional) Run test kitchen against the tags `ListCommand`,`SearchCommand` and `InfoCommand`

### Operating Systems Testing

- Windows 11
- Windows Server 2019

## Change Types Made


## Change Checklist


## Related Issue

- Fixes #158
- https://app.clickup.com/t/20540031/PROJ-526
